lsix

```diff
@@ -182,7 +182,46 @@ autodetect() {
     numtiles=$((width/(tilewidth + 2*tilexspace + 1)))
 }
 
+declare -a FILE_LIST
+
+# filter_filename adds filename to FILE_LIST if it  is supported by imagemagick
+# adds :----[0] tag to specify first image for multiple images formats
+filter_filename_and_specify_first_image() {
+    local file="$1"
+    local number_of_frames
+    number_of_frames="$(identify -quiet -format "%n" "$file" 2> /dev/null)"
+    [ "$?" -eq 1 ] && return # Skip file if not recognized
+    if [ "$number_of_frames" -gt 1 ] ; then
+        FILE_LIST+=( ":${file}[0]" )
+    else
+        FILE_LIST+=( "$file" )
+    fi
+}
+
+# build_filelist builds the list of files to be displayed
+# It mimics the behaviour of 'ls' for directories.
```

Comment:
Why mimic ls? Why not simply use ls or bash wildcards?

Reply:
I though the goal of lsix was to behaves like ls as its name implies.
I used find because I need the full path for build the filename list.
I didn't found any usefull option in ls to have full pathname.

Reply:
You are correct, one of `lsix`'s goals is to behave like `ls`. However, a higher goal is simplicity so it will be easily hackable by other people to suit their preferences. 

If we can figure out a way to do the same thing in one line by leveraging bash, I'd prefer that. Perhaps by checking each command line argument to see if it is a directory and then recursively calling lsix like so: `cd $directory; $0; cd -`

Reply:
> If we can figure out a way to do the same thing in one line by leveraging bash

I disagree with this one. I found hard to read one-liners because the implies multiple concern at once.
More over, with `cd $directory; $0; cd -` if it fails between cds, you are returning in another folder than the one you launch the command.

Reply:
Good point about one-liners being hard for some people to read. I find them easier when it is doing something extremely simple or to make error handling not break up the flow.

Also, it's a good point about cd possibly failing. I've handled that by making it even shorter: `(cd $directory; $0)`. (Well, essentially that, check out my latest code for specifics).

Reply:
I just check out your master code and give it a try.
The filename sort is right for me so I think that the `Sort order is a bit strange.` comment can be removed.

Incidentally, adding the output "Recursing on ... " fixes the mlterm bug :)

But when threre are folder empty of recognized extensions, it is a little verbose to have these outputs without images:

```
Recursing on /storage/owncloud/fredo/Images/2018/02/03
Recursing on /storage/owncloud/fredo/Images/2018/02/06
Recursing on /storage/owncloud/fredo/Images/2018/02/09
```

Good job !